src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -278,6 +290,82 @@ public int GetHashCode(Build obj)
             }
         }
 
+        /// <summary>
+        ///     Given a downloaded build, determine what the product name should be
+        ///     for the release json
+        /// </summary>
+        /// <param name="build">Downloaded build</param>
+        /// <returns>Product name</returns>
+        private string GetProductNameForReleaseJson(DownloadedBuild build)
+        {
+            // Preference the github repo name over the azure devops repo name.
+            if (!string.IsNullOrEmpty(build.Build.GitHubRepository))
+            {
+                // Split off the github.com+org name and just use the repo name, all lower case.
+                (string owner, string repo) = GitHubClient.ParseRepoUri(build.Build.GitHubRepository);
+                return repo.ToLowerInvariant();
+            }
+            else if (!string.IsNullOrEmpty(build.Build.AzureDevOpsRepository))
+            {
+                // Use the full repo name without project/account
+                (string accountName, string projectName, string repoName) = AzureDevOpsClient.ParseRepoUri(build.Build.AzureDevOpsRepository);
+                return repoName.ToLowerInvariant();
+            }
+            else
+            {
+                throw new NotImplementedException("Unknown repository name.");
+            }
+        }
+
+        /// <summary>
+        ///     Given a downloaded build, determine what the fileshare location should be.
+        /// </summary>
+        /// <param name="build">Downloaded build</param>
+        /// <returns>File share location</returns>
+        public string GetFileShareLocationForReleaseJson(DownloadedBuild build)
+        {
+            // We only want to have shipping assets in the release json, so append that path
+            return Path.Combine(build.OutputDirectory, shippingSubPath);
+        }
+
+        /// <summary>
+        ///     Write the release json.  Only applicable for separated (ReleaseLayout) drops
+        /// </summary>
+        /// <param name="downloadedBuilds">List of downloaded builds</param>
+        /// <param name="outputDirectory">Output directory write the release json</param>
+        /// <returns>Async task</returns>
+        private async Task WriteReleaseJson(List<DownloadedBuild> downloadedBuilds, string outputDirectory)
+        {
+            if (_options.DryRun || !_options.ReleaseLayout)
+            {
+                return;
+            }
+
+            Directory.CreateDirectory(outputDirectory);
+            string outputPath = Path.Combine(outputDirectory, "release.json");
+
+            var releaseJson = new[]
+            {
+                new
+                {
+                    release = _options.ReleaseName,
+                    products = new[]
+                    {
+                        downloadedBuilds.Where(b => b.AnyShippingAssets)
```

Comment:
This array of one enumerable should just be the enumerable. (Mentioned the effect on the json in IM but might as well comment.)